### ISSUE TITLE
Set up continuous integration for automatic deployment

### ISSUE THREAD

**Issue by dandavies99** [ImperialCollegeLondon/gridlington-vis#21]
The ultimate goal is to have one image that is pulled automatically to the VM from the GitHub Container Registry when changes in the repo are pushed to `main`. 

There are several aspects to think about for this:
- Containerising everything that is not already an image pulled from elsewhere for OVE, including:
  - The dash app
  - The config files required by the other images (in `/config/)
- Ensuring that the correct host machine IP address is present in all the necessary files.
  - For the development server this should never change
  - For local development, this will change frequently

Aha! Link: https://imperial-college-ict.aha.io/requirements/RSE-140-2

**Comment by dandavies99** [comment]
I discovered that several of the config files are not read/needed unless using the OVE asset manager. So `config/` can be simplified to: 

```
config/
  - default.conf
  - Spaces.json
```

And everything still works. This means the `configure.py` script can also be simplified to only edit the `docker-compose.yml` file.

My plan was to `COPY` the `config` directory into the `dash` service within the Dockerfile, e.g. `COPY ./config/ ./config`, which in itself works. Then to modify the `docker-compose.yml` to make each of the two files available to `nginx` and `ovehub-ove-ove` by mounting them as shared volumes. 

`docker-compose build` works with the below docker-compose.yml file, but `docker-compose up` fails with `Error response from daemon: source /var/lib/docker/overlay2/<id>/merged/config/default.conf is not directory`, which is not a very helpful message. 

```yaml
services:
  dash:
    build: .
    ports:
      - 8050:8050
    volumes:
      - nginx-default-conf:/config/default.conf ## New
      - ove-spaces-conf:/config/Spaces.json   ## New
  nginx:
    depends_on:
      - dash
      - ovehub-ove-ove
      - ovehub-ove-apps
      - ovehub-ove-service-persistence-inmemory
      - ovehub-ove-ui
    image: nginx:alpine
    ports:
      - 8080:8080
    restart: unless-stopped
    volumes:
      - nginx-default-conf:/etc/nginx/conf.d/default.conf:ro ## New
      - ./logs/nginx.log:/etc/nginx/logs/error.log
  openvidu-openvidu-call:
    environment:
      openvidu.publicurl: https://146.169.222.39:4443
      openvidu.secret: MY_SECRET
    image: openvidu/openvidu-call:2.12.0
    ports:
      - 4443:4443
    restart: unless-stopped
  ovehub-ove-apps:
    depends_on:
      - ovehub-ove-ove
      - ovehub-ove-external-tuoris
      - openvidu-openvidu-call
    environment:
      OPENVIDU_HOST: 146.169.222.39:4443
      OPENVIDU_SECRET: MY_SECRET
      OVE_HOST: 146.169.222.39:8080
      TOURIS_HOST: 146.169.222.39:7080
      TUORIS_HOST: 0.0.0.0:7080
    image: ovehub/ove-apps:0.5.0
    restart: unless-stopped
  ovehub-ove-external-tuoris:
    image: ovehub/ove-external-tuoris:v0.1
    ports:
      - 7080:7080
    restart: unless-stopped
  ovehub-ove-ove:
    depends_on:
      - dash
    image: ovehub/ove:0.5.0
    restart: unless-stopped
    volumes:
      - ove-spaces-conf:/usr/src/app/packages/ove-core/dist/client/Spaces.json ## New
  ovehub-ove-service-persistence-inmemory:
    image: ovehub/ove-service-persistence-inmemory:0.2.1
    restart: unless-stopped
  ovehub-ove-ui:
    depends_on:
      - ovehub-ove-ove
      - ovehub-ove-apps
    environment:
      OVE_HOST: 146.169.222.39:8080
    image: ovehub/ove-ui:0.2.0
    restart: unless-stopped
version: '3.1'
volumes:
  nginx-default-conf:  ## New
  ove-spaces-conf:    ## New

```

If we can get this working then the containers would be self-sufficient and we could just push new versions of the dash image to the VM. Any ideas welcomed.

**Comment by AdrianDAlessandro** [comment]
I believe the main questions in this have been addressed in #24 and implementing/finalising the changes can wait until development on the vis system is begun in earnest.
